x/archivedb/key_value_reader.go

```diff
@@ -82,13 +30,48 @@ func (reader *dbHeightReader) Has(key []byte) (bool, error) {
 }
 
 // Get retrieves the given key if it's present in the key-value data store.
+//
+// This is a public API, so the dbKey is used to retrieve any value. It is
+// guaranteed to be a O(1), because how the dbKey is structured internally which
+// prevents a longer key to match a shorter key with the same prefix.
+//
+// If the result matches a dbMetaKey it will be consired as a ErrNotFound
+// because dbMetaKey is for internal usage and should be leaked outside of the
+// module
 func (reader *dbHeightReader) Get(key []byte) ([]byte, error) {
-	iterator, err := reader.getIteratorToValue(key)
+	iterator := reader.db.rawDB.NewIteratorWithStart(newDBKey(key, reader.height))
 	defer iterator.Release()
+
+	reader.heightLastFoundKey = 0
+
+	if !iterator.Next() {
+		if err := iterator.Error(); err != nil {
+			return nil, err
+		}
+
+		// There is no available key with the requested prefix
+		return nil, database.ErrNotFound
+	}
+
+	foundKey, height, err := parseDBKey(iterator.Key())
```

Comment:
Tried to write a unit test for this and realized that we actually can't hit an error other than `ErrNotFound` here because `binary.ReadUvarint` won't error, `reader.Read` will not error, and `len([]byte("archivedb.height")) > minInternalKeyLen`.

That being said, I think we can make this code simpler.